tools/rules_config/_core.sh

```diff
@@ -21,6 +21,30 @@ debug() {
   fi
 }
 
+### Read FILEPATH_SONAR_RULES_REUSED file to extract RULES_KEYS
+function read_sonar_rules_reused {
+  debug "Read file ${FILEPATH_SONAR_RULES_REUSED}"
+  linenumber=1
+  interal_field_separator='|'
+  rules_keys=""
+  {
+    #Skip header file markdown
+    read -r
+    read -r
+    while IFS='|' read -ra line; do
+      col=0;
+      for substr in "${line[@]}"; do
+        if [ $col -eq 1 ]; then
+          rules_keys="$rules_keys$(echo $substr | xargs),"
+        fi
+        col=$((col+1))
+      done
+    done
+  } < $FILEPATH_SONAR_RULES_REUSED
+  debug "Extracting rules: ${rules_keys::-1} \n"
```

Comment:
on my local machine (MAC OSx), I have an error on lines with "::-1" string : 
`_core.sh: line 44: -1: substring expression < 0`

thus my questions :
- why not use ":-1" string value instead of "::-1" ?
- why do you want to affect the default value `1` to rules_key ? for me no pb if no value in this variable. this use case is controlled => we will have an error like `ERROR - config parameter 'RULES_KEYS' is NOT valid : it can't be empty`

Reply:
I concatenate each keys with a comma. The use of operator `::-1` is here to delete the last comma i have added.

I have tested for the experience:
```
  debug "Extracting rules: ${rules_keys::-1} \n"
  RULES_KEYS=${rules_keys:-1}
```
```
--- DEBUG --- Read file ./SONAR_RULES_REUSED.md
--- DEBUG --- Extracting rules: css:S4655,php:S2014,Web:ItemTagNotWithinContainerTagCheck

--- DEBUG --- SONAR_TOKEN : xxxxxxxxxxxxxxxxxxxxxxxxxxxxx
--- DEBUG --- SONAR_URL : http://localhost:9000
--- DEBUG --- RULES_KEYS : css:S4655,php:S2014,Web:ItemTagNotWithinContainerTagCheck,
--- DEBUG --- TAG_ECOCONCEPTION : test1
```
You will see the comma not deleted with `:-1`. 
i find why you have a bug.... It's a function from bash 4.2 and you must have bash 3 if i'm looking the readme :D [delete-the-last-character-of-a-string-using-string-manipulation-in-shell-script](https://unix.stackexchange.com/questions/144298/delete-the-last-character-of-a-string-using-string-manipulation-in-shell-script)
I will look for an older method.

Reply:
I use ZSH to launch scripts ... here are my versions (ZSH but also BASH if you want)

```shell
~ ❯ zsh --version
zsh 5.9 (x86_64-apple-darwin22.0)
~ ❯ bash --version
GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin22)
Copyright (C) 2007 Free Software Foundation, Inc.
~ ❯ 

```